### ISSUE TITLE
Double indentation for script tag in html

### ISSUE THREAD

**Issue by lobre** [neovim/neovim#15423]
### Neovim Version

v0.5.0

### Operating system/version:

Ubuntu 20.04

### Terminal name/version:

xfce4-terminal

### TERM environment variable

xterm-256color

### Installation

appimage

### Steps to reproduce

I am starting from a simple HTML `index.html` file containing only a pair of `<html>` tags.

```html
<html>
</html>
```

I am using neovim v0.5.0, downloaded from the appimage, and I am opening the above HTML file with:

```
nvim.appimage -u NORC index.html
```

Then, I put my cursor on the opening tag `<html>` and go to insert mode at the next line using `o`. Then I start to type `<script`. But when I add the closing character `>`, the tag gets pushed one level of indentation to the right.

To me, this is not supposed to happen. And if I repeat the exact same steps with neovim v0.4.4, I don’t have the problem. The `<script>` stays where it is.

See the problem in action.

![nvim_script_indent](https://user-images.githubusercontent.com/3764103/129983265-924b1a27-1309-41fa-9f2e-87b290f2409b.gif)

I don’t know if it can help, but there were 4 commits altering `runtime/indent/html.vim` between v0.4.4 and v0.5.0.

```
$ git log --oneline v0.4.4..v0.5.0 -- runtime/indent/html.vim
2081504a3 vim-patch:942db23c9cb7
b1cd67b62 vim-patch:4072ba571bab
27fb03f5a vim-patch:7ff78465f705
7c6ca6fed vim-patch:73fef33014db
```

### Vim

Bug happens with vim -u DEFAULTS

### Expected behavior

When closing the tag with `>`, I am expecting to have this output (`<script>` does not move, and stay at its level of indentation).

```html
<html>
        <script>
</html>
```

### Actual behavior

Instead, I have this output (with 2 indentation levels).

```html
<html>
                <script>
</html>
```

**Comment by justinmk** [comment]
> Bug happens with vim -u DEFAULTS

That indicates it's a runtime / syntax file question, or perhaps related to a default setting. Not a bug, but a usage question.

**Comment by lobre** [comment]
@justinmk, I am not sure to understand why you say it indicates it’s a question and not a bug? What am I supposed to do with this?

I can understand that it is not a bug in neovim "core", but it is still a bug of syntax/indentation that does not occur in v0.4.4. Where am I supposed to report that?

Thanks

**Comment by clason** [comment]
> I can understand that it is not a bug in neovim "core", but it is still a bug of syntax/indentation that does not occur in v0.4.4. Where am I supposed to report that?

The procedure with syntax file issues is to check the offending file for the maintainer and contact them. Their fix should then be included in the vim runtime files, from which neovim will pull them.

In this case, the file is `runtime/indent/html.vim`, which says `Maintainer: Bram Molenaar`. So you can skip the first step and report it directly to https://github.com/vim/vim.

**Comment by lobre** [comment]
Thanks @clason for explaining the workflow.

After browsing vim issues, I can see the problem has already been reported here.

https://github.com/vim/vim/issues/7215

And a patch has already been applied.

> from which neovim will pull them.

Is that an automatic process? If so, at which frequency are runtime files pulled from vim? Or should someone also apply the patch as a PR in neovim?

**Comment by clason** [comment]
Can you link that patch? It's not obvious from the issue (and you _did_ mention that vim behaves the same?)

No, it's not automatic, and the process depends on whether it's a "proper" patch or a runtime file update.

**Comment by clason** [comment]
OK, the change seems to have been made in the June 13 runtime file update (which we haven't merged yet). Can you confirm that vim 8.2.2993 or newer behaves correctly?

**Comment by lobre** [comment]
> and you did mention that vim behaves the same?

I understand the confusion now, and why @justinmk closed the issue in the first place. I did not understand correctly this part from the PR template: `vim -u DEFAULTS`. I thought it was to verify that Neovim did not have the issue with the default config while the question about Vim itself (my `vim` to `nvim` alias is the culprit...).

I just tested with vim 8.2.3360 and **it behaves correctly**.

And yes, the patch is not obvious as it is not properly linked to the issue, but it seems to correspond to that commit:

https://github.com/vim/vim/commit/2346a6378483c9871016f9fc821ec5cbea638f13
